combin/combin.go

```diff
@@ -0,0 +1,105 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package combin implements routines involving combinatorics (permutations,
+// combinations, etc.).
+package combin
+
+import "math"
+
+const (
+	badNegInput = "combin: negative input"
+	badSetSize  = "combin: n < k"
+)
+
+// Binomial returns the binomial coefficient of (n,k).
+//
+// The binomial coefficient is the number of unordered combinations of k elements
+// in a set that is n elements big. It is also commonly referred to as "n choose k".
+//
+//  n!/((n-k)!k!)
+//
+// n and k must be non-negative with n >= k, otherwise Binomial will panic.
+// No check is made for overflow.
+func Binomial(n, k int) int {
+	if n < 0 || k < 0 {
+		panic(badNegInput)
+	}
+	if n < k {
+		panic(badSetSize)
+	}
+	// (n,k) = (n, n-k)
+	if k > n/2 {
+		k = n - k
+	}
+	kfact := 1
```

Comment:
R language uses floats for [n choose k](https://github.com/wch/r-source/blob/b156e3a711967f58131e23c1b1dc1ea90e2f0c43/src/nmath/choose.c#L110), which then allows:
```
res = n;
for i := 2; i <= k; i++ {
	    res *= (n-i+1)/i;
}
```
For k up to 29.  Not sure what source they are using for this code though.